File: ios/ios-reference/App/App/Plugins/SaveTransferPlugin.swift
import Foundation
import Capacitor

@objc(SaveTransferPlugin)
public final class SaveTransferPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "SaveTransferPlugin"
    public let jsName = "SaveTransfer"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getBridgeInfo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "pickImport", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "exportDocument", returnType: CAPPluginReturnPromise),
    ]

    @objc public func getBridgeInfo(_ call: CAPPluginCall) {
        call.resolve([
            "plugin": "SaveTransfer",
            "bridgeVersion": 1,
            "platform": "ios",
            "availability": "available",
            "implementation": "skeleton",
            "capabilities": ["pick_import", "export_document"]
        ])
    }

    @objc public func pickImport(_ call: CAPPluginCall) {
        rejectNotImplemented(call, method: "pickImport")
    }

    @objc public func exportDocument(_ call: CAPPluginCall) {
        rejectNotImplemented(call, method: "exportDocument")
    }

    private func rejectNotImplemented(_ call: CAPPluginCall, method: String) {
        call.reject("SaveTransfer.\(method) is not implemented in Phase 04.", "NOT_IMPLEMENTED")
    }
}

File: android-reference/app/src/main/java/com/ceegore/riftwarden/plugins/SaveTransferPlugin.java
package com.ceegore.riftwarden.plugins;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SaveTransfer")
public final class SaveTransferPlugin extends Plugin {
    private static final int BRIDGE_VERSION = 1;

    @PluginMethod
    public void getBridgeInfo(PluginCall call) {
        JSObject result = new JSObject();
        result.put("plugin", "SaveTransfer");
        result.put("bridgeVersion", BRIDGE_VERSION);
        result.put("platform", "android");
        result.put("availability", "available");
        result.put("implementation", "skeleton");
        result.put("capabilities", new String[] { "pick_import", "export_document" });
        call.resolve(result);
    }

    @PluginMethod
    public void pickImport(PluginCall call) { rejectNotImplemented(call, "pickImport"); }

    @PluginMethod
    public void exportDocument(PluginCall call) { rejectNotImplemented(call, "exportDocument"); }

    private void rejectNotImplemented(PluginCall call, String method) {
        call.reject("SaveTransfer." + method + " is not implemented in Phase 04.", "NOT_IMPLEMENTED");
    }
}
